scripts/ci/test_plan.py

```diff
@@ -196,7 +196,7 @@ def find_archs(self):
 
                 self.get_plan(_options, True)
             else:
-                self.get_plan(_options, False)
+                self.get_plan(_options, True)
```

Comment:
Is it really correct?

As I understand new logic - if we found changes in some architecture code we currently limits testing on this architecture to integration mode.

That's probably fine for some architectures we commonly have in `integration_platforms` list in tests - but it not so fine for other architectures.

And we don't want to add additinal platforms to `integration_platforms` in tests - otherwise it will increase test execution time for all other runs (where architecture code isn't changed).

CC: @abrodkin @ruuddw

Reply:
the goal of all of this is to reduce the CI noise and build only for what has changed, it is not about increasing coverage if some architecture code has changed. This target PRs and help with reducing CI runtime.